cli/tests/test_next_prev_commands.rs

```diff
@@ -567,6 +567,127 @@ fn test_next_editing() {
     "###);
 }
 
+#[test]
+fn test_prev_conflict() {
+    // Make the third commit our new parent.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    std::fs::write(&file_path, "Other text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    // Create a conflict in the third commit, where we'll jump to.
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(second)"]);
+    std::fs::write(&file_path, "some text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(third)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "fourth"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["prev", "--conflict"]);
+    // We now should be a child of `third`.
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: vruxwmqv 61428b9f (conflict) (empty) (no description set)
+    Parent commit      : kkmpptxz 9de45eb7 (conflict) third
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+     "###);
+}
+
+#[test]
+fn test_prev_conflict_editing() {
+    // Edit the third commit.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    std::fs::write(&file_path, "Other text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    // Create a conflict in the third commit, where we'll jump to.
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(second)"]);
+    std::fs::write(&file_path, "some text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(third)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "fourth"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["prev", "--conflict", "--edit"]);
+    // We now should be editing the third commit.
```

Comment:
Since there's a conflict in the fourth commit and in the working copy, I would expect us to be editing the fourth commit here. I suppose we ended up editing the third commit because the fourth commit was empty, but that's a bit confusing to me.

Reply:
I removed the fourth commit and filtered out empty commits for `prev --conflict`..